router.go

```diff
@@ -601,6 +625,21 @@ func getVersion(c echo.Context) error {
 	doc.Rev = ""
 	doc.Attachments = nil
 
+	// Check if we have the local tarfile
+
+	conf, err := config.GetConfig()
+	if err != nil {
+		return err
+	}
+	sc := conf.SwiftConnection
+
+	base := filepath.Base(doc.URL)
+	fp := filepath.Join(appSlug, version, base)
+	buf := new(bytes.Buffer)
+	if _, err := sc.ObjectGet(prefix, fp, buf, false, nil); err == nil {
+		updatedURL := fmt.Sprintf("%s://%s/%s/registry/%s/%s/tarball/%s", c.Scheme(), c.Request().Host, space.Prefix, appSlug, version, base)
+		doc.URL = updatedURL // Update the URL to point to the registry instead of the downcloud/aws/... url
+	}
```

Comment:
Why don't we put the updated URL in CouchDB directly after saving this file in swift? It would avoid to open a connection to swift here.